src/test/xmeta/test_symbol_table.cpp

```diff
@@ -12,6 +12,614 @@
+        : id{ name }, sem{ sem }, return_type{ return_type }, params{ params } {}
+
+    void VerifyType(std::shared_ptr<method_model> const& actual)
+    {
+        REQUIRE(actual->get_id() == id);
+        REQUIRE(actual->get_semantic().is_static == sem.is_static);
+        REQUIRE(actual->get_return_type()->get_semantic().is_resolved());
+
+        if (return_type == std::nullopt)
+        {
+            REQUIRE(actual->get_return_type() == std::nullopt);
+        }
+        else
+        {
+            REQUIRE(actual->get_return_type()->get_semantic().is_resolved());
+            return_type->VerifyType(*actual->get_return_type());
+        }
+
+        auto const& actual_params = actual->get_formal_parameters();
+        REQUIRE(actual_params.size() == params.size());
+        for (size_t i = 0; i < params.size(); i++)
+        {
+            params[i].VerifyType(actual_params[i]);
+        }
+    }
+};
+
+struct ExpectedPropertyModel
+{
+    std::string id;
+    property_semantics sem;
+    ExpectedTypeRefModel type;
+
+    ExpectedMethodModel get_method;
+    std::optional<ExpectedMethodModel> set_method;
+
+    ExpectedPropertyModel(std::string const& name,
+        property_semantics const& sem,
+        ExpectedTypeRefModel const& type,
+        ExpectedMethodModel const& get_method,
+        std::optional<ExpectedMethodModel> const& set_method)
+        : id{ name }, sem{ sem }, type{ type }, get_method{ get_method }, set_method{ set_method } {}
+
+    void VerifyType(std::shared_ptr<property_model> const& actual)
+    {
+        REQUIRE(actual->get_id() == id);
+        REQUIRE(actual->get_semantic().is_static == sem.is_static);
+        REQUIRE(actual->get_type().get_semantic().is_resolved());
+        type.VerifyType(actual->get_type());
+
+        REQUIRE(actual->get_get_method());
+        get_method.VerifyType(actual->get_get_method());
+
+        if (actual->get_set_method())
+        {
+            set_method->VerifyType(actual->get_set_method());
+        }
+        else
+        {
+            REQUIRE(set_method == std::nullopt);
+        }
+    }
+};
+
+struct ExpectedEventModel
+{
+    std::string id;
+    event_semantics sem;
+    ExpectedTypeRefModel type;
+
+    ExpectedMethodModel add_method;
+    ExpectedMethodModel remove_method;
+
+    ExpectedEventModel(std::string const& name,
+        event_semantics const& sem,
+        ExpectedTypeRefModel const& type)
+        : id{ name }, sem{ sem }, type{ type }
+    {
+        method_semantics method_sem;
+        if (sem.is_static)
+        {
+            method_sem.is_static = true;
+        }
+        add_method = { "add_" + id, method_sem, ExpectedTypeRefModel{ ExpectedStructRef{ "Foundation.EventRegistrationToken" } }, {
+            ExpectedFormalParameterModel{ "TODO:findname", parameter_semantics::in, type } } };
```

Comment:
TODO?

Reply:
Yea I didn't see the spec didn't specifiy a name for the parameter